File: modules/lua_juce_core/containers/Array.cpp
#include "Array.hpp"

auto juce_Array(sol::table& state) -> void
{
    juce_ArrayImpl<int>(state, "Array_int");
    juce_ArrayImpl<float>(state, "Array_float");
    juce_ArrayImpl<double>(state, "Array_double");

    juce_ArrayImpl<juce::String>(state, "Array_String"); // StringArray is a separate type.
    juce_ArrayImpl<juce::File>(state, "Array_File");
    juce_ArrayImpl<juce::var>(state, "Array_var");
}

File: test/core/Array.lua
local int = juce.Array_int.new()
assert(int:isEmpty())
assert(int:size() == 0)
int:add(42)
assert(not int:isEmpty())
assert(int:size() == 1)
int:resize(143)
assert(int:size() == 143)

local float = juce.Array_float.new()
assert(float:isEmpty())
assert(float:size() == 0)

local double = juce.Array_double.new()
assert(double:isEmpty())
assert(double:size() == 0)

local String = juce.Array_String.new()
assert(String:isEmpty())
assert(String:size() == 0)

local File = juce.Array_File.new()
assert(File:isEmpty())
assert(File:size() == 0)

local var = juce.Array_var.new()
assert(var:isEmpty())
assert(var:size() == 0)
